app/src/main/java/com/hover/stax/utils/StaxVolleySingleton.java

```diff
@@ -0,0 +1,166 @@
+package com.hover.stax.utils;
+
+import android.annotation.SuppressLint;
+import android.content.Context;
+import android.content.pm.ApplicationInfo;
+import android.content.pm.PackageManager;
+import android.net.ConnectivityManager;
+import android.net.NetworkInfo;
+import android.util.Log;
+
+import com.android.volley.AuthFailureError;
+import com.android.volley.DefaultRetryPolicy;
+import com.android.volley.Request;
+import com.android.volley.RequestQueue;
+import com.android.volley.Response;
+import com.android.volley.NetworkResponse;
+import com.android.volley.ParseError;
+import com.android.volley.toolbox.HttpHeaderParser;
+import com.android.volley.toolbox.HurlStack;
+import com.android.volley.toolbox.JsonArrayRequest;
+import com.android.volley.toolbox.JsonObjectRequest;
+import com.android.volley.toolbox.RequestFuture;
+import com.android.volley.toolbox.StringRequest;
+import com.android.volley.toolbox.Volley;
+import com.hover.sdk.BuildConfig;
+import com.hover.sdk.R;
+import com.hover.sdk.security.KeyHelper;
+import com.hover.sdk.utils.AnalyticsSingleton;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import java.io.UnsupportedEncodingException;
+
+import java.io.IOException;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.TimeoutException;
+
+final public class StaxVolleySingleton {
```

Comment:
No, don't do this for something so small. Isn't there a simple way to add a header to an okhttp request?

Reply:
It's ok to add it to the default so that it is in every request